### PR TITLE
Improve UX for editing component translations

### LINKED ISSUE #2478: FormIO translations for components refactor
Currently, FormIO translations are stored globally on the `FormDefinition`, but to display them the FormIO builder injects these translations into the components. The issues with this are:
- we essentially have two states that contain the information
- Because the schema is immutable, any changes to the translations do not show up in other components until the `FormDefinition` is saved

It would be better if the `state.componentTranslations` were leading, and the components parse these to determine which translations are relevant. Additionally, there should be an onChange event to update this `state.componentTranslations`

Additionally, the translatable fields should be prefilled to remove the need for copy pasting

### PR DESCRIPTION
**Pull request by stevenbal**
Fixes #2478 

TODO:
- [x] implement prefilling
  - [x] prefill labels of `TRANSLATABLE_FIELDS`
    - [x] issue with datefield placeholder?
    - [x] showing datefield defaultValue makes not sense (same for numberfield and others, there should probably a mapping that determines the relevant translatable fields per component? or an override per component?)
  - [x] in case label changes, change it in the translations as well (form-creation-form.js)
    - [x] what if original label is reused in other component, but the label is changed in one of them -> original label shouldn't be changed but new one should be added
    - [x] what if two labels are different and one is changed so they become the same? -> the translation of the changed literal is used
- [x] how to deal with content field, no prefilling needed?
- [x] ensure updates to literals leads to updated translations immediately (before saving, relies on `EDIT_STEP`)
  - seems that edit_step only has the full config (not updated) and triggers way too often 
  - customDefaultValue only runs on opening the modal
  - cannot really update the step.componentTranslations state, because there is a chance the changes aren't saved
- [x] consistent ordering between languages
- [x] make literals readonly in translations? -> doesn't seem to work for datagrid
- [x] when saving, remove empty translations
- ~~implement deleting?~~
- ~~getValuesForField -> why does it return arrays?~~
- [x] general code cleanups
- [x] cleanup unused translations?
- [x] check if layout components are not skipped

**Test/edge cases**

- [x] Entering translation for a literal, then use the same literal in another field
- [x] Enter translations, edit literal -> keep the original translation
- [x] Enter translations -> edit `key` property -> keep the original translation
- [x] Add Selenium integration test to ensure translations can be entered and viewed
    - Explore if we can get drag and drop working (!)
    - Otherwise just start with a configuration and edit that

### CODE REVIEW

File: src/openforms/forms/forms/form_definition.py [hunk 1]
```diff
@@ -21,5 +21,6 @@ class Meta:
             "login_required",
             "is_reusable",
             "configuration",
+            "component_translations",
```
Comment (sergei-maertens) [inline src/openforms/forms/forms/form_definition.py:24]: TODO -> this isn't going right now in the admin

File: src/openforms/js/components/form/translations.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import {Formio} from 'react-formio';
+
+import {DEFAULT_TEXT_TABS} from './edit/tabs';
+
+class TranslationsDataGrid extends Formio.Components.components.datagrid {
```
Comment (sergei-maertens) [inline src/openforms/js/components/form/translations.js:5]: TODO: check if we can't enforce this behaviour via options

File: src/openforms/js/components/formio_builder/index.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ onLoaded(() => {
   [...FORM_BUILDERS].forEach(node => {
     const configurationInput = BEM.getChildBEMNode(node, BLOCK_FORM_BUILDER, INPUT_ELEMENT);
     const configuration = JSON.parse(configurationInput.value) || {display: 'form'};
+    // FIXME: handle translations?
```
Comment (sergei-maertens) [inline src/openforms/js/components/formio_builder/index.js:17]: TODO: should be possible to manage component translations from standalone admin page too

File: src/openforms/js/formio_module.js [hunk 4]
```diff
@@ -60,6 +61,7 @@ const FormIOModule = {
     currency: CurrencyField,
     editgrid: EditGrid,
     datamap: Datamap,
+    translationsDataGrid: TranslationsDataGrid,
```
Comment (sergei-maertens) [inline src/openforms/js/formio_module.js:64]: TODO: check if we can avoid this

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
Selenium tests are flaky still - old translations may show up

**Comment by sergei-maertens** [comment]
Ok, all that's left to do is cleaning up the commit history if CI passes

**Comment by sergei-maertens** [comment]
image

**Review by sergei-maertens — APPROVED** [review]
(just approving to get rid of the 'changes requested' status)

File: src/openforms/js/components/formio_builder/translation.js [hunk 5]
```diff
@@ -1,3 +1,254 @@
...
+  fieldset: ['legend'],
+  editgrid: ['groupLabel'],
+  // manually entered data values -> labels are translatable
+  select: [/data\.values\[[0-9]+\]\.label/],
+  selectboxes: [/values\[[0-9]+\]\.label/],
+  radio: [/values\[[0-9]+\]\.label/],
+};
+
+const isRegex = item => {
+  return typeof item.test === 'function';
```
Comment (CharString) [inline src/openforms/js/components/formio_builder/translation.js:47]: I wouldn't be surprised if String were function too in javascript xD

File: src/openforms/js/components/formio_builder/translation.js [hunk 6]
```diff
@@ -1,3 +1,254 @@
...
+ * of a particular component.
+ *
+ * @param  {Object} container Key-value mapping, key is language code, values are mappings
+ *                            of literals to their translations *for that language code*.
+ * @param  {Object} component Formio component/schema - it contains the configuration
+ *                            of one particular component in the form builder.
+ * @return {Void}             The container is mutated directly. It is advised to pass
+ *                            in an Immer draft or carefully manage mutable objects.
+ */
+export const persistComponentTranslations = (container, component) => {
```
Comment (CharString) [inline src/openforms/js/components/formio_builder/translation.js:97]: Reading `persistComponentTranslations` elsewhere without context, made me think persistence to the backend. While it's actually moving data from the passed in component to the container... Can't really think of a better name :thinking:

File: src/openforms/js/components/formio_builder/translation.spec.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    expect(isTranslatableProperty('textfield', 'defaultValue')).toBe(true);
+    expect(isTranslatableProperty('date', 'defaultValue')).toBe(false);
+  });
+
+  it('works with nested paths', () => {
+    expect(isTranslatableProperty('textfield', 'data.values[3].label')).toBe(false);
+    expect(isTranslatableProperty('select', 'data.values[2].value')).toBe(false);
+    expect(isTranslatableProperty('select', 'data.values[1].label')).toBe(true);
+    expect(isTranslatableProperty('radio', 'values[420].label')).toBe(true);
+    expect(isTranslatableProperty('selectboxes', 'values[420].label')).toBe(true);
```
Comment (CharString) [inline src/openforms/js/components/formio_builder/translation.spec.js:28]: Yep. What were they smoking?

File: src/openforms/scss/components/builder/_builder.scss [hunk 8]
```diff
@@ -59,6 +59,9 @@ div.flatpickr-calendar.open {
   .tab-pane.active {
     min-height: 50vh;
   }
+  .formio-component-content .tab-pane.active {
+    min-height: 0;
```
Comment (CharString) [inline src/openforms/scss/components/builder/_builder.scss:63]: I don't even want to know...

File: src/openforms/forms/tests/selenium_tests/test_form_designer.py [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+
+from furl import furl
+from selenium.common import NoSuchElementException
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+from openforms.accounts.tests.factories import SuperUserFactory
+from openforms.tests.selenium_tests.base import SeleniumTestCase
```
Comment (CharString) [inline src/openforms/forms/tests/selenium_tests/test_form_designer.py:15]: `base` is a first in the codebase. But "utils" feels off too... Maybe move it to __init__.py. Meh. Base it is...
